build.cake

```diff
@@ -37,39 +38,72 @@ Task("Clean")
-              var buildNumber = AppVeyor.Environment.Build.Number;
-              packageVersion = version + "-CI-" + buildNumber + dbgSuffix;
-              if (AppVeyor.Environment.PullRequest.IsPullRequest)
-                  packageVersion += "-PR-" + AppVeyor.Environment.PullRequest.Number;
-              else if (AppVeyor.Environment.Repository.Branch.StartsWith("release", StringComparison.OrdinalIgnoreCase))
-                  packageVersion += "-PRE-" + buildNumber;
-              else
-                  packageVersion += "-" + AppVeyor.Environment.Repository.Branch;
-          }
-
-          AppVeyor.UpdateBuildVersion(packageVersion);
-      }
+        NuGetRestore(SOLUTION);
     });
 
 //////////////////////////////////////////////////////////////////////
+// INITIALIZE BUILD
+//////////////////////////////////////////////////////////////////////
+
+Setup(context =>
+{
+    if (IsRunningOnWindows())
+    {
+        var settings = new GitVersionSettings
+        {
+            OutputType = GitVersionOutput.Json,
+        };
+
+        /*if (!BuildSystem.IsLocalBuild)
+        {
+            settings.LogFilePath = "console";
+            settings.OutputType = GitVersionOutput.BuildServer;
+        }*/
+
+        var gitVersion = GitVersion(settings);
+
+        Information("AssemblySemVer: " + gitVersion.AssemblySemVer);
+        Information("BranchName: " + gitVersion.BranchName);
+        Information("BuildMetaData: " + gitVersion.BuildMetaData);
+        Information("BuildMetaDataPadded: " + gitVersion.BuildMetaDataPadded);
+        Information("CommitDate: " + gitVersion.CommitDate);
+        Information("CommitsSinceVersionSource: " + gitVersion.CommitsSinceVersionSource);
+        Information("CommitsSinceVersionSourcePadded: " + gitVersion.CommitsSinceVersionSourcePadded);
+        Information("FullBuildMetaData: " + gitVersion.FullBuildMetaData);
+        Information("FullSemVer: " + gitVersion.FullSemVer);
+        Information("InformationalVersion: " + gitVersion.InformationalVersion);
+        Information("LegacySemVer: " + gitVersion.LegacySemVer);
+        Information("LegacySemVerPadded: " + gitVersion.LegacySemVerPadded);
+        Information("Major: " + gitVersion.Major);
+        Information("Minor: " + gitVersion.Minor);
+        Information("NuGetVersion: " + gitVersion.NuGetVersion);
+        Information("NuGetVersionV2: " + gitVersion.NuGetVersionV2);
+        Information("Patch: " + gitVersion.Patch);
+        Information("PreReleaseLabel: " + gitVersion.PreReleaseLabel);
+        Information("PreReleaseNumber: " + gitVersion.PreReleaseNumber);
+        Information("PreReleaseTag: " + gitVersion.PreReleaseTag);
+        Information("PreReleaseTagWithDash: " + gitVersion.PreReleaseTagWithDash);
+        Information("SemVer: " + gitVersion.SemVer);
+        Information("Sha: " + gitVersion.Sha);
+
+        packageVersion = gitVersion.NuGetVersion ?? context.EnvironmentVariable("GitVersion_NuGetVersion");
```

Comment:
We had some logic in the original script that did different things based on whether there was a tag and the name of the branch. Is that all incorporated somewhere or are we getting default settings now?